### PR TITLE
feat: add /species/combine controller

### PR DESCRIPTION
**Pull request by luacmartins**
This PR is related to the combine species issue on the client side and creates the /species/combine endpoint.

### CODE REVIEW

File: src/controllers/species.controller.ts
Comment (dadiorchen) [inline src/controllers/species.controller.ts]: Seems with Loopback, is easy to wrap the executions in a transaction, for this `for` loop, it might bring some safety. 
Or, could we think about use bulk update to reduce the connection count to the DB? Like: `update set a=x where id in (1,2,3)`

**Review by dadiorchen — APPROVED** [review]
Another question, how do we verify it works? Make up some data in DB and request the endpoint manually?

File: src/controllers/species.controller.ts
Comment (luacmartins) [reply src/controllers/species.controller.ts]: That sounds interesting! I will try to implement the bulk update version. Thanks for the feedback.

For the testing, I was using it in parallel with the client to manually add/combine species and also using Postman. Do we have any other way to test it?

File: src/controllers/species.controller.ts
Comment (nmcharlton) [inline src/controllers/species.controller.ts]: Unfortunately, this won't work in test and production environments, due to deletes being prohibited (issue #486). We should really set `active=false`.

**Review by nmcharlton — CHANGES_REQUESTED** [review]
It might be worth fixing #486 at the same time to ensure consistent delete behaviour.

File: src/controllers/species.controller.ts
Comment (luacmartins) [reply src/controllers/species.controller.ts]: Makes sense. Although active is an integer column. Are we using 0 for inactive and 1 for active?

Comment (nmcharlton) [reply src/controllers/species.controller.ts]: I think that must have been implemented incorrectly - it should be a Boolean. We'll need to raise a bug against the database project ([treetracker-database](https://github.com/Greenstand/treetracker-database)) and create a database migration script to correct it.

Comment (luacmartins) [reply src/controllers/species.controller.ts]: I talked to @deepwinter and filed an issue there. I'm sending another commit with the updated code but we should wait until changes are made to the database to merge it.

**Review by nmcharlton — APPROVED** [review]
Lovely work 👍

**Pull request merged**
